File: articles/history-of-decision-science/python/run_all_history_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for The History of Decision Science."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "expected_value_historical_baseline.py",
    "expected_utility_bernoulli_model.py",
    "subjective_probability_savage_model.py",
    "bayesian_update_history_example.py",
    "bounded_rationality_satisficing.py",
    "behavioral_noise_choice_model.py",
    "minimax_regret_history.py",
    "robustness_historical_paradigms.py",
    "decision_record_exporter.py",
    "history_of_decision_science_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python history workflows completed.")


if __name__ == "__main__":
    main()

File: articles/history-of-decision-science/r/run_all_history_workflows.R
# run_all_history_workflows.R
# Run all R workflows for The History of Decision Science.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "history_of_decision_science_workflow.R",
  "expected_utility_history_profiles.R",
  "subjective_probability_profiles.R",
  "regret_and_robustness_history.R",
  "bounded_rationality_summary.R",
  "historical_rank_visualization.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R history workflows completed.")
